byzcoin/contracts.go

```diff
@@ -38,19 +39,111 @@ type Contract interface {
 	Delete(ReadOnlyStateTrie, Instruction, []Coin) ([]StateChange, []Coin, error)
 }
 
+// ReadOnlyContractRegistry is the read-only interface for the contract registry.
+type ReadOnlyContractRegistry interface {
+	Search(contractID string) (ContractFn, bool)
+}
+
+// SpawnerContract is an interface to detect contracts that need a reference
+// to the registry.
+type SpawnerContract interface {
```

Comment:
Not convinced by the name. Spawning is the only use case when a contract needs the registry?

Reply:
Ahah me neither. I tried to find a name for a contract that needs to look up other contracts but miserably failed. Something like `ContractWithRegistry` sounds weird but at least it's clear, I don't know.

Reply:
I didn't find any better name either, I would go for `ContractWithRegistry` as it's at least clear.